driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraphFactory.java

```diff
@@ -54,4 +100,40 @@ protected Configuration getConfiguration(boolean create, boolean open, boolean t
         }};
     }
 
+    /**
+     * @param create if true automatically creates database if database with given URL does not exist
+     * @param open   if true automatically opens the database
+     */
+    protected ODatabaseDocumentTx getDatabase(boolean create, boolean open) {
+        final ODatabaseDocumentTx db = new ODatabaseFactory().createDatabase("graph", url);
+        if (!db.getURL().startsWith("remote:") && !db.exists()) {
+            if (create) db.create();
+            else if (open) throw new ODatabaseException("Database '" + url + "' not found");
+        } else if (open) db.open(user, password);
+
+        return db;
+    }
+
+    /**
+     * Setting up the factory to use database pool instead of creation a new instance of database connection each time.
+     */
+    public OrientGraphFactory setupPool(final int max) {
+        pool = new OPartitionedDatabasePool(url, user, password, max).setAutoCreate(true);
```

Comment:
Would be also nice, to make this option available using configuration...

That way, or tests can randomly include it and so some tests would run under pool and others w/o pool.